### ISSUE TITLE
Download registered as completed, but hash check returned unfinished chunks.

### ISSUE THREAD

**Issue by FlyingPersian** [rakshasa/rtorrent#287]
Hi
I get this error once in a while in rtorrent: "Download registered as completed, but hash check returned unfinished chunks.". I had this on a torrent from one site, then downloaded it from a different site and had the issue again. I rehash the torrent, it shows this error then and a short while after goes as complete. 
I'm running rtorrent 0.9.4 with libtorrent 0.13.4 on FreeNAS (FreeBSD).

**Comment by rakshasa** [comment]
Likely a bug in the kernel's handling of mmap'ed memory sync'ing to file.

This has been known to happen to old/forked *BSD kernels and even exposed a bug in linux's handling of >4GB mmap'ed files.

**Comment by FlyingPersian** [comment]
I noticed that when I disabled the run of a script, nzbToMedia, on complete this problem didn't occur:

I had two big files that gave me the error, even when I tried to download the remaining 1,5% multiple times. I disabled the run of the script on complete, retried downloading the remaining 1,5% et voila, it worked. 
At some point those rars were fine, I could extract them manually, but when nzbToMedia copied the rars to a temp folder those rars could not be extracted. So I assumed that the way they're copied was bad, so I changed it from hardlinking to symboliclinking to copying, neither worked at first. I then sticked with copy, which seemed to work best.
